cvat/apps/dataset_manager/tests/_test_formats.py

```diff
@@ -401,6 +402,22 @@ def load_dataset(src):
                     self.assertEqual(len(dataset), task["size"])
                 self._test_export(check, task, format_name, save_images=False)
 
+    def test_can_skip_outside(self):
+        images = self._generate_task_images(3)
+        task = self._generate_task(images)
+        self._generate_annotations(task)
+        task_ann = TaskAnnotation(task["id"])
+        task_ann.init_from_db()
+        task_data = TaskData(task_ann.ir_data, Task.objects.get(pk=task["id"]))
+
+        extractor = CvatTaskDataExtractor(task_data, include_outside=False)
+        dm_dataset = datumaro.components.project.Dataset.from_extractors(extractor)
+        self.assertEqual(4, len(dm_dataset.get("image_1").annotations))
+
+        extractor = CvatTaskDataExtractor(task_data, include_outside=True)
```

Comment:
I believe we always should include_outside but different formats can use or ignore such shapes.

Reply:
Maybe, but there is no formats with such annotation property.

Reply:
We've been there already. It means that all formats must expect (read as support) such shapes.

Reply:
Now "caller" knows about format implementation details (a format doesn't support "outside" flag). Is it OK? What is about attributes? Tracks? Cuboids? Are you going to filter that as?

Reply:
I think, only CVAT knows about `outside`. Tracks and cuboids are not exported too.

Reply:
OK. I see that CvatTaskDataExtractor is used as an utility function by each format. Thus the knowledge encapsulated.